[ADD] accounting/l10n_us: usa accounting

samueljlieber opened the pull request:
Task: [#3912161](https://www.odoo.com/web?debug=1#id=3912161&cids=3&menu_id=4720&model=project.task&view_type=form)
Parent task: [#3900288](https://www.odoo.com/web?debug=1#id=3900288&cids=3&menu_id=4720&action=333&active_id=3835&model=project.task&view_type=form)

Adds USA l10n accounting doc to the fiscal localizations.

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/9206).

StraubCreative commented:
5/6: Confirmed with DAK that the COA list was accurate / looks ok 👍 
Need to check with DEV on **Accounting Reports** and **Cash Discounts** sections.

StraubCreative commented:
5/7 PHT feedback:
> CoA, clients would like to know more about sub accounts.
> Reporting side, clients would like to know more details about cash flow statement.
> I have a forum post talking about more cash flow statement if you are curious:
> https://www.odoo.com/forum/help-1/v16-how-does-odoo-generate-cash-flow-statement-operating-financing-investing-activities-235233

StraubCreative commented:
PHT initial feedback implemented on [87f0b68](https://github.com/odoo/documentation/pull/9206/commits/87f0b6851c40cd7de696fb2fd2dcdac726e7778e). All commits squashed to this point to include the initial commits for overview, config, CoA, reports, and avatax sections.

StraubCreative commented:
CSV edit on [fbb5409](https://github.com/odoo/documentation/pull/9206/commits/fbb540985f741e83fe0f71e67414009d658a50ce) (removed BE lines). 

PR is ready for collaboration @tiku-odoo @larm-odoo

StraubCreative commented:
Squashed all commits on [014b374](https://github.com/odoo/documentation/pull/9206/commits/014b3743f56ff63075541930135ff89135775293). 

Sections to do still:
- cash discount
- taxes
- writing checks

samueljlieber commented:
Made the following updates in e0710de :

- Added all `l10n_us` modules to the **Modules** section and added internal :ref:s on the modules name to each relevant section
- Added content to the **Taxes** section, and moved **AvaTax** section below the **Taxes** section
- Added content to the **Cash discounts** section which mainly links to the _Cash discount_ documentation
- Changed **ACH** section title to "ACH - electronic transfers"
- Fixed typos and RST syntax issues
- Added/fixed :menuselection: use in a few navigation instructions
- Added the `coa-account-types.png` image to the **View, edit, and sort accounts** section (there was a comment to add this image)
- Renamed all image files to not have underscores

---

The following sections still need content:
- **Writing checks**
- **1099 form**

samueljlieber commented:
Made the following updates in a9f0011:
- Added internal references to each reporting section
- Added Check Register and 1099 Report to the **Reporting** section
- Moved **1099 Report** section under the **Reporting** section, and added 1099 content
- Added content to the **Writing checks** section and linked to existing Pay by checks documentation
---
At this point all sections have content– need to review the sections I added for completeness and accuracy.

samueljlieber commented:
Made the following updates in 2a02bd8:

- Simplified the **Chart of Accounts** section to only list the predefined COA for l10n_us in a tip admonition
- Full technical review (and fixes) of all RST, images, and links
- Squashed all commits

samueljlieber commented:
688baad  Made the edits detailed in this task: https://www.odoo.com/odoo/project/1625/tasks/3977110?cids=3

StraubCreative commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,746 @@
+First, ensure the *United States - Payroll - Export to ADP* (`l10n_us_hr_payroll_adp`) module is
+:ref:`installed <general/install>`.
+
+Then, the company **must** have an *ADP Code* entered for the company settings. To do so, navigate
```
```suggestion
Then, the company **must** have an *ADP Code* entered in the company settings. To do so, navigate
```

StraubCreative commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,746 @@
+Export work entries to ADP
+--------------------------
+
+Requirements
```
For future improvements:
Think it's worth it to further explain each code to differentiate them more for the reader (is it *one* ADP code or is it *three* or more than that?), as far as what each code is/means and where to find them in ADP. A screenshot or two would be useful here too, given how important these steps.

StraubCreative commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,746 @@
+~~~~~~~~~~~
+
+Once :doc:`work entries <../../hr/payroll/work_entries>` have been verified, the information can be
+exported into a CSV file, which can then be uploaded into ADP.
```
```suggestion
exported to a CSV file, which can then be uploaded into ADP.
```

StraubCreative commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,746 @@
+Export`, then click :guilabel:`New`. Next, enter the :guilabel:`Start Date` and :guilabel:`End Date`
+for the work entries using the calendar pop-over.
+
+Then, enter a :guilabel:`Batch ID` in the corresponding field. The recommendation for this field is
+to enter the date in a `YY-MM-DD` format, followed by any other characters to distinguish that
+specific batch, such as a department name, or any other defining characteristics for the batch.
+
+Enter a :guilabel:`Batch Description` in the corresponding field. This should be short and
+descriptive, but distinct from the :guilabel:`Batch Name`.
```
Screenshot would be helpful here too, to provide a visual model for people to reference.

StraubCreative commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,746 @@
+<../../essentials/search>` to more easily find the employees to add to the list.
+
+.. tip::
+   It is helpful to process the data export in multiple groups instead of in one large group
+   containing all employees. The most common ways to group employees is by department or by wage
+   type (hourly or salaried).
```
Context
```suggestion
   Process the data export in multiple groups instead of in one large group that contains all
   employees. This helps to meaningfully differentiate the batches and makes processing more
   tenable, overall. The most common ways to group employees is by department, or by wage type
   (hourly or salaried).
```

StraubCreative commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,746 @@
+
+Automated Clearing House (ACH) payments are a modern way to transfer funds electronically between
+bank accounts, replacing traditional paper-based methods. |ACH| payments are commonly used for
+direct deposits, bill payments, and business transactions, offering convenience and
+cost-effectiveness. With their widespread adoption, |ACH| payments have become an integral part of
+managing finances securely and efficiently.
```
Don't need sales-y language
```suggestion
direct deposits, bill payments, and business transactions.
```

StraubCreative commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,746 @@
+   institution once they have been approved to send payments via their account.
+
+Under the section labeled, :guilabel:`NACHA configuration` are the fields required to generate a
+|NACHA| compatible |ACH| file to send to a company's bank. First, enter the *routing number* of the
```
Don't need the emphasis here (not Odoo product or field name)
```suggestion
|NACHA| compatible |ACH| file to send to a company's bank. First, enter the routing number of the
```

StraubCreative commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,746 @@
+:guilabel:`Destination`. This information will be provided by the bank or credit union.
+
+Following the :guilabel:`Destination` field is the :guilabel:`Immediate Origin` field. Enter the
+9-digit company ID (Employer Identification Number (EIN)) into this field. This information is
```
to cover cases where EIN isn't available (such as sole proprietorship with SSN) or TIN.

```suggestion
9-digit company ID or Employer Identification Number (EIN) into this field. This information is
```

StraubCreative commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,746 @@
+9-digit company ID (Employer Identification Number (EIN)) into this field. This information is
+provided by the financial institution.
+
+Next, the :guilabel:`Company Identification` number should be entered. This is a 10-digit number
+made up of the 9-digit company ID (Employer Identification Number (EIN)), with an additional number
+at the start of the sequence. This number is often a `1`. Check with the financial institution
+should this first number differ to verify that it is correct. This number is provided for |ACH|
+approved accounts.
```
Remove passive voice
```suggestion
Next, enter the :guilabel:`Company Identification` number, which is a 10-digit number made from
combining the 9-digit company ID or Employer Identification Number (EIN), along with an additional
number at the start of the sequence. This number is often a `1`. Check with the financial
institution should this first number differ to verify that it is correct, as this number is provided
for |ACH| approved accounts.
```

StraubCreative commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,746 @@
+should this first number differ to verify that it is correct. This number is provided for |ACH|
+approved accounts.
+
+The :guilabel:`Originating DFI Identification` number is next, this Depository Financial Institution
+(DFI) field contains an assigned 8-digit number. Consult the financial institution for this number.
```
Condense
```suggestion
Enter the :guilabel:`Originating DFI Identification` number next, which should contain an assigned
8-digit number from the financial institution.
```

StraubCreative commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,746 @@
+(DFI) field contains an assigned 8-digit number. Consult the financial institution for this number.
+
+.. important::
+   Enter the above numbers exactly as the company's financial institution (bank or credit union) has
+   provided them. Failing to do so, would be detrimental to the successful  |NACHA| configuration.
```
Wording
```suggestion
   Enter the numerical values in this section *exactly* as the company's financial institution
   (e.g. bank or credit union) has provided them, otherwise risk failing a successful |NACHA|
   configuration in Odoo.
```

StraubCreative commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,746 @@
+of the field to enable :guilabel:`Generated Balanced Files`. Consult the company's accountant or
+financial advisor to make an informed decision for this field.
+
+Manually :icon:`fa-cloud-upload` :guilabel:`(cloud upload)` save the configuration or navigate away
```
Wording, to more clearly tie the result to the clickable action
The prior version places the icon label next to the main action verb, which risks confusing readers.

```suggestion
Manually save the configuration by clicking the :icon:`fa-cloud-upload` :guilabel:`(cloud upload)` icon, or navigate away
```

StraubCreative commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,746 @@
+   :align: center
+   :alt: The exported file tab highlighted in the batch payment with the download circled.
+
+If any adjustments need to be made, click the :guilabel:`Re-gererate Export File` button to recreate
```
typo
```suggestion
If any adjustments need to be made, click the :guilabel:`Re-generate Export File` button to recreate
```

StraubCreative commented:
@robodoo r+

dade-odoo commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,748 @@
+         selected.
+
+.. warning::
+   All payments in the batch must share the same payment method.
```
As mentioned above, it's not enough for these to just be the same payment method, they must all be the NACHA payment method.

dade-odoo reviewed (commented):
Hey @samueljlieber @StraubCreative @larm-odoo 
I know I'm late to the party here, but excellent job on this localization! I can see how much work y'all put into it because I'm struggling to find much to say about it. Most of my comments ended up being more format-related than accounting-related, so take those suggestions as you will.
There were just a few accounting-related comments that I think would be great to include. Please let me know if you have any questions and thanks again for knocking this thing out of the park!

samueljlieber commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,748 @@
+     - Adds United States accounting reports.
+   * - :guilabel:`US Checks Layout`
+     - `l10n_us_check_printing`
+     - Enables the printing of payments on pre-printed check paper. Supports the three most common
```
Awesome thanks! Noted in 17.3 l10n_us IMP task: https://www.odoo.com/odoo/project.task/3979114

samueljlieber commented on content/applications/finance/fiscal_localizations/united_states.rst:
```diff
@@ -0,0 +1,748 @@
+From the :guilabel:`Chart of Accounts` dashboard, create new accounts by clicking the purple
+:guilabel:`New` button in the top-left corner of the dashboard and :ref:`filling in the
+corresponding form <chart-of-account/create>`. Search and sort through existing accounts by using
+specific :guilabel:`Filters` and :guilabel:`Group By` criteria, which are available in the mega menu
```
I agree we should stick wit ha more common term, we have been using "drop-down menu" for the "search drop-down menu"
